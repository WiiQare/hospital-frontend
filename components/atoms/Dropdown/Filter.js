import { HiChevronDown } from "react-icons/hi";
import RangeSlider from "../Input/RangeSlider";

const DropdownFilter = ({
  label,
  icon,
  items,
  className,
  labelClassName,
  chevronDown = true,
  dropClassName,
  onClick = () => {},
}) => {
  return (
    <div className={`${className} dropdown dropdown-end flex`}>
      <label
        tabIndex={0}
        className={` bg-white border rounded-xl w-full py-2 px-4 flex justify-between items-center ${labelClassName}`}
      >
        <span className="flex mr-2 gap-4 items-center">
          {typeof icon == "function" ? icon() : icon}
          <span className="font-light">{label}</span>
        </span>
        {chevronDown ? <HiChevronDown size={20} /> : <></>}
      </label>
      <ul
        tabIndex={0}
        className={`${dropClassName} w-full mt-14 dropdown-content menu p-2 bg-white border rounded-box text-gray-700 font-light`}
      >
        {items ? (
          items.map((item, index) => (
            <li onClick={onClick} key={index}>
              <a>{item}</a>
            </li>
          ))
        ) : (
          <RangeSlider
            initialMin={10}
            initialMax={10000}
            min={10}
            max={10000}
            step={1}
            priceCap={1000}
          />
        )}
      </ul>
    </div>
  );
};

export default DropdownFilter;
