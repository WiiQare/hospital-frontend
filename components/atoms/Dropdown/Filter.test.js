import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Filter from "./Filter";

describe("Filter", () => {
  let component;
  const label = "Filter";
  beforeEach(() => {
    const res = render(<Filter label={label} />);
    component = res.container;
  });

  it("should render the component", () => {
    expect(component).toMatchSnapshot();
  });

  test("should render text elements", () => {
    expect(screen.getByText(label)).toBeInTheDocument();
  });
});
