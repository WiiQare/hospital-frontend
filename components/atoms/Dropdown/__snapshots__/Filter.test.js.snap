// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should render the component 1`] = `
<div>
  <div
    class="undefined dropdown dropdown-end flex"
  >
    <label
      class=" bg-white border rounded-xl w-full py-2 px-4 flex justify-between items-center undefined"
      tabindex="0"
    >
      <span
        class="flex mr-2 gap-4 items-center"
      >
        <span
          class="font-light"
        >
          Filter
        </span>
      </span>
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="20"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
          fill-rule="evenodd"
        />
      </svg>
    </label>
    <ul
      class="undefined w-48 mt-14 dropdown-content menu p-2 bg-white border rounded-box text-gray-700 font-light"
      tabindex="0"
    >
      <div
        class="overflow-hidden"
      >
        <div
          class="flex justify-between items-center my-6 "
        >
          <div
            class="rounded-md flex items-center"
          >
            <span
              class="p-2 font-extralight text-xs"
            >
               Min
            </span>
            <label
              class="text-sky font-bold text-2xl flex items-start"
              for=""
            >
              10
               
              <span
                class="text-gray-500 font-light text-xs"
              >
                $/mo
              </span>
            </label>
          </div>
          <div
            class="ml-2 font-semibold text-lg"
          >
             - 
          </div>
          <div
            class=" "
          >
            <div
              class="rounded-md flex items-center"
            >
              <span
                class="p-2 font-extralight text-xs"
              >
                 Max
              </span>
              <label
                class="text-sky font-bold text-2xl flex items-start"
                for=""
              >
                10000
                <span
                  class="text-gray-500 font-light text-xs"
                >
                  $/mo
                </span>
              </label>
            </div>
          </div>
        </div>
        <div
          class="mb-4"
        >
          <div
            class="slider relative h-1 rounded-md bg-gray-300"
          >
            <div
              class="progress absolute h-1 bg-green-300 rounded "
              style="left: 0.001%; right: 0%;"
            />
          </div>
          <div
            class="range-input relative  "
          >
            <input
              class="range-min absolute w-full  -top-1  h-1   bg-transparent  appearance-none pointer-events-none"
              max="10000"
              min="10"
              step="1"
              type="range"
              value="10"
            />
            <input
              class="range-max absolute w-full  -top-1 h-1  bg-transparent appearance-none  pointer-events-none"
              max="10000"
              min="10"
              step="1"
              type="range"
              value="10000"
            />
          </div>
        </div>
      </div>
    </ul>
  </div>
</div>
`;
